src/http.rs

```diff
@@ -53,17 +54,25 @@ where
     pub fn new(
         tls_settings: impl Into<MaybeTlsSettings>,
         proxy_config: &ProxyConfig,
+        timeout_duration: Option<Duration>,
     ) -> Result<HttpClient<B>, HttpError> {
-        HttpClient::new_with_custom_client(tls_settings, proxy_config, &mut Client::builder())
+        HttpClient::new_with_custom_client(
+            tls_settings,
+            proxy_config,
+            timeout_duration,
+            &mut Client::builder(),
+        )
     }
 
     pub fn new_with_custom_client(
         tls_settings: impl Into<MaybeTlsSettings>,
         proxy_config: &ProxyConfig,
+        timeout_duration: Option<Duration>,
         client_builder: &mut client::Builder,
     ) -> Result<HttpClient<B>, HttpError> {
         let mut http = HttpConnector::new();
         http.enforce_http(false);
+        http.set_connect_timeout(timeout_duration.or_else(|| Some(Duration::from_secs(5))));
```

Comment:
I'd prefer this number be given a `const` near the top of this file to make it a little more explicit and easier to recognize if it needs changing.